cmd/kubeadm/app/cmd/upgrade/plan_test.go

```diff
@@ -599,31 +617,26 @@ Note: Before you can perform this upgrade, you have to update kubeadm to v1.8.3.
 
 _____________________________________________________________________
 
+No information available on component configs.
 `,
 		},
 	}
 
 	for _, rt := range tests {
 		t.Run(rt.name, func(t *testing.T) {
 			rt.buf = bytes.NewBufferString("")
-			outputFlags := newUpgradePlanPrintFlags(rt.outputFormat)
+			outputFlags := output.NewOutputFlags(&upgradePlanTextPrintFlags{}).WithTypeSetter(outputapischeme.Scheme).WithDefaultOutput(rt.outputFormat)
 			printer, err := outputFlags.ToPrinter()
 			if err != nil {
 				t.Errorf("failed ToPrinter, err: %+v", err)
 			}
 
-			// Generate and print upgrade plans
-			for _, up := range upgrades {
-				plan, unstableVersionFlag, err := genUpgradePlan(&up, rt.externalEtcd)
-				if err != nil {
-					t.Errorf("failed genUpgradePlan, err: %+v", err)
-				}
-				printUpgradePlan(&up, plan, unstableVersionFlag, rt.externalEtcd, rt.buf, printer)
-			}
+			plan := genUpgradePlan(upgrades, nil, false)
```

Comment:
(same here)
instead of nil for the second arg, please have at least one test case where the component configs are available.